Automattic/wp-calypso
# Detect swipe gestures in DotPager to hide popovers in containing Tasks components

#### Changes proposed in this Pull Request

![Screen Shot 2022-02-09 at 1 00 18 PM](https://user-images.githubusercontent.com/797888/153125122-1eb4ff0e-4b5f-486d-b610-812602e716bf.png)

This goal of this PR is to fix a mobile issue in My Home. The primary `DotPager` of the page contains many `Task` components, and many of those has a button "Hide this" which triggers a popover menu with options. An issue occurs when the user's swipe gesture doesn't properly hide the popover menu, resulting in the page moving but the popover menu remains static.

To solve this issue, an `isSwiping` state was added to the `Swipeable` component which `DotPager` uses as dependency to detect swipe gestures. In addition, two properties `onPageSwipeStart` and `onPageSwipeEnd` were also added to the component so that the parent component (i.e.: `DotPager`) would be aware that a swipe gesture is taking place.

Powered with this knowledge, the `DotPager` is now capable of passing down the state to its children (`Task`), and in turn the `Task` component will hide the `PopoverMenu` component whenever user starts swiping.

#### Testing instructions



1. Start by opening the browser's developer tools, toggle the device toolbar and make the browser viewport mobile.
2. Head to My Home.
3. Locate the primary `DotPager` (shown in the screenshot at the top of the PR), and find a card with the button "Hide this".
4. Click on "Hide this" to trigger the `PopoverMenu` with options.
5. Swipe the page, and expect the `PopoverMenu` to close.
6. (Optional) Swipe to a next/previous page and repeat steps 4-5.



Related to #60905
